### ISSUE TITLE
Dialog results

### ISSUE THREAD

**Issue by sabuto** [amwx/FluentAvalonia#93]
Is there a way to get custom results from dialogs? instead of just none, primary or secondary? for example if i was doing a custom dialog with a form, i would want the form data back?

**Comment by amwx** [comment]
ContentDialogs really just manage their existence, not the content within. It is just a ContentControl, so you can treat it like any other. Set up a ViewModel that represents your form, assign that to the Content property, then set the DataTemplate to build your form. Then when the dialog returns, you just read your ViewModel to get the data you want. 

```C#
var formViewModel = new MyFormViewModel();

var cd = new ContentDialog
{
    Content = formViewModel ,
    ContentTemplate = new FuncDataTemplate((x, ns) => 
    {
         return new MyForm(); // Custom UserControl representing your form
    },
    PrimaryButtonText = "OK",
    CloseButtonText = "Cancel"
};

var result = await cd.ShowAsync();
if (result == ContentDialogResult.Primary)
{
    // handle the data within formViewModel
}
```
Of course ContentDialogs can be declared in Xaml so if you don't like doing it in C# you can set it up there instead and just use `this.FindControl<ContentDialog>("MyDialog")` to retrieve it.

**Comment by Maruhl** [comment]
I would also be interested in a solution where you could define the buttons yourself with text and maybe icons. In principle the ContentDialog would only need a ButtonResult-Collection or similar in which you can decide freely instead of using the fixed properties PrimaryButtonText, etc..

**Comment by nlogozzo** [comment]
Can be closed as #104 has merged
